File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnConfig.java

package com.gracenote.gnsdk;

public class GnConfig extends GnConfigObject {
  private transient long swigCPtr;

  protected GnConfig(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnConfig_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnConfig obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnConfig(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public GnConfig() throws com.gracenote.gnsdk.GnException {
    this(gnsdk_javaJNI.new_GnConfig__SWIG_0(), true);
  }

  public GnConfig(String config) throws com.gracenote.gnsdk.GnException {
    this(gnsdk_javaJNI.new_GnConfig__SWIG_1(config), true);
  }

  public void set(String custom, String value) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnConfig_set__SWIG_0(swigCPtr, this, custom, value);
  }

  public void set(GnConfigOptionAccess access) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnConfig_set__SWIG_1(swigCPtr, this, access.swigValue());
  }

  public void set(GnConfigOptionEnable enable, boolean value) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnConfig_set__SWIG_2(swigCPtr, this, enable.swigValue(), value);
  }

  public void set(GnConfigOptionLocation location, String value) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnConfig_set__SWIG_3(swigCPtr, this, location.swigValue(), value);
  }

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnConfig.cs

namespace GracenoteSDK {

public class GnConfig : GnConfigObject {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;

  internal GnConfig(global::System.IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnConfig_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnConfig obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnConfig() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnConfig(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public GnConfig() : this(gnsdk_csharp_marshalPINVOKE.new_GnConfig__SWIG_0(), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public GnConfig(string config) : this(gnsdk_csharp_marshalPINVOKE.new_GnConfig__SWIG_1(config), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public void Set(string custom, string value) {
    gnsdk_csharp_marshalPINVOKE.GnConfig_Set__SWIG_0(swigCPtr, custom, value);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public void Set(GnConfigOptionAccess access) {
    gnsdk_csharp_marshalPINVOKE.GnConfig_Set__SWIG_1(swigCPtr, (int)access);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public void Set(GnConfigOptionEnable enable, bool value) {
    gnsdk_csharp_marshalPINVOKE.GnConfig_Set__SWIG_2(swigCPtr, (int)enable, value);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public void Set(GnConfigOptionLocation location, string value) {
    gnsdk_csharp_marshalPINVOKE.GnConfig_Set__SWIG_3(swigCPtr, (int)location, value);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

}

}
